packages/dotcom-ui-header/src/components/dropdown-navigation/dropdownNavigation.js

```diff
@@ -1,93 +1,208 @@
+  const response = await fetch(url)
+  if (!response.ok) {
+    throw new Error(`HTTP error! Status: ${response.status}`)
+  }
+  return response.json()
+}
 
-    // When clicking outside the dropdown, reset the dropdown button by
-    // removing the data attribute from the dropdown button to enable the dropdown to be opened correctly again
-    document.addEventListener('click', (event) => {
-        dropdowns.forEach((dropdownContainer) => {
-            const dropdownButton = dropdownContainer.querySelector('.o-header__dropdown-button');
-            if (!dropdownContainer.contains(event.target)) {
-                dropdownButton.removeAttribute('data-dropdown-button-active');
-            }
-        });
-    });
-
-    // Observe when the sticky header becomes visible and when it hides
-    // and when it does we need to close the dropdown if it is open
-    // This is needed since there are two headers and each has a dropdown
-    // so we shouldn't be showing the sticky header's dropdown when the
-    // sticky header is not visible, or show the regular header's dropdown
-    // when the sticky header is visible
-    const stickyHeader = document?.querySelector('.o-header--sticky')
-    const stickyHeaderObserver = new IntersectionObserver(
-        (changes) => {
-            for (let change of changes) {
-                if (
-                    document.documentElement.clientWidth >= 780 ||
-                    (document.documentElement.clientWidth < 780 && !change.isIntersecting)
-                ) {
-                    const closestDropdownParent = document.activeElement.closest('.o-header__dropdown')
-                    if (closestDropdownParent) {
-                        const dropdownButton = closestDropdownParent.querySelector('.o-header__dropdown-button')
-                        dropdownButton.removeAttribute('data-dropdown-button-active')
-                        document.activeElement.blur()
-                    }
-                }
-            }
-        },
-        { threshold: [0.01] }
-    )
-
-    stickyHeaderObserver.observe(stickyHeader)
+const updateLinksList = (dropdown, links, trackingKey) => {
+  const list = dropdown.querySelector('.o-header__dropdown-list')
+  if (!list) {
+    console.error('Links list element not found')
+    return
+  }
+
+  // Determine the list item template
+  const label = dropdown.querySelector('.o-header__dropdown-list-item-label-container')
+  const listItem = label?.closest('.o-header__dropdown-list-item') || list.firstElementChild
+
+  if (!listItem) {
+    console.error('List item template not found')
+    return
+  }
+
+  // Build the updated list
+  const updatedListFragment = document.createDocumentFragment()
+  links.forEach((link) => {
+    const updatedItem = buildListItem(listItem, label, link, trackingKey)
+    updatedItem && updatedListFragment.append(updatedItem)
+  })
+
+  // Update the DOM
+  list.innerHTML = ''
+  list.append(updatedListFragment)
```

Comment:
possible convenience method?
```suggestion
  list.replaceChildren(updatedListFragment)
```

Reply:
`replaceChildren` is safe to use in modern browsers from 2020 onwards. Can we afford this? 
We went with backward compatibility for browsers from around 2017-2019,  which is why we avoided relatively modern CSS features like :has().

Reply:
Thanks for the context @pdimova . It does make sense to try to support these older browsers where possible. However, based on our [browser support matrix/policy,](https://docs.google.com/document/d/1z6kecy_o9qHYIznTmqQ-IJqre72jhfd0nVa4JMsS7Q4/edit?usp=sharing) I think 2020+ features fit perfectly. I don't have any strong preference either way. `replaceChildren` just looked more convenient and might have a more performant implementation than first clearing the node and then doing an `append`.